beacon-chain/db/iface/interface.go

```diff
@@ -99,7 +99,7 @@ type HeadAccessDatabase interface {
 	SaveHeadBlockRoot(ctx context.Context, blockRoot [32]byte) error
 
 	// Genesis operations.
-	LoadGenesis(ctx context.Context, r io.Reader) error
+	LoadGenesis(ctx context.Context, stateBytes []byte) error
```

Comment:
@rauljordan what do you think about changing from reader to byte slice? I recall you had some feedback about this in the past.

Reply:
Adding some context here - I generally really like the idea of Reader, but when downloading from network the reader becomes a little bit awkward. `http.Client` relies on `response.Body` being `Close()`d for connection reuse. Requiring bytes instead simplifies things at the source, because the value can be read to a buffer and passed as a slice. Otherwise there is design overhead in deciding where the `ReadCloser` should get `Close`d and making sure it happens. I have thought about making a type that wraps a `ReaderCloser` as a `Reader`, automatically calling `Close()` when the last byte has been read. We could try that out in a follow-up. I think Readers could be of benefit at the p2p -> blockchain level, ensuring we never buffer things to memory until needed. But even there, we often need to read the entire ssz object to unmarshal it for validation.

Reply:
Another thought - if ssz consumed/produced readers instead of byte slices, there would be more value in using readers, because the reader could incrementally read to/from the network as the ssz code does its job. But since ssz wants a complete byte slice, we ultimately will `ioutil.ReadAll` the reader anyway, once it becomes time to deal with it.

Reply:
One last thought - now that we have the `state.BeaconState` and `block.SignedBeaconBlock` interfaces, perhaps the db package shouldn't deal with byte level representations at all? it could be the job of the `Initializer` to go from `[]byte`->`state.BeaconState`, which would be consistent with other db methods that deal in higher-level types.